packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -225,8 +226,16 @@ async function getIslandContext (event: H3Event): Promise<NuxtIslandContext> {
   return ctx
 }
 
+const HAS_APP_TELEPORTS = appTeleportTag && appTeleportId
```

Comment:
It would be worth checking whether this correctly tree-shakes when these are unset, if you haven't already.